README.md

```diff
@@ -137,16 +137,16 @@ async def main(subnet_tag="testnet"):
     # By passing `event_emitter=log_summary()` we enable summary logging.
     # See the documentation of the `yapapi.log` module on how to set
     # the level of detail and format of the logged information.
-    async with Engine(
+    async with Executor(
         package=package,
         max_workers=3,
         budget=10.0,
         timeout=init_overhead + timedelta(minutes=len(frames) * 2),
         subnet_tag=subnet_tag,
         event_emitter=log_summary(),
-    ) as engine:
+    ) as executor:
 
-        async for task in engine.map(worker, [Task(data=frame) for frame in frames]):
+        async for task in executor.submit(worker, [Task(data=frame) for frame in frames]):
```

Comment:
I'd dispute `submit` as the right choice ... the task is not only submitted for execution here but rather, it's submitted, executed, results are awaited for and the execution only yields to this loop once the particular fragments are already completed ... so actually, maybe `run` would be a better choice ...

Reply:
or actuallty, `execute` for that matter ... although `Executor.execute` does sound off ...

Reply:
Well, actually `executor.submit()` (or `engine.map()`, for that matter) actually just submits the task for execution, and returns asynchronous iterator over completed tasks. It can be used like this:
```
results = executor.submit(worker, tasks)
# at this point, some work may have been done on the providers, or may not 
# and now we wait for the results:
for task in results:
    ...
```

Reply:
In this respect, `Engine.map()` (aka `Executor.submit()`) is really similar to `Executor.submit()` from Python's `concurrent.futures`, which schedules a function to be executed and returns a future.